version/version.go

```diff
@@ -1,4 +1,119 @@
 package version
 
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
 // Version is the version of the build.
 const Version = "1.14.1-dev"
+
+// crioVersion is a type for parsing and writing a VersionFile for CRI-O
+type crioVersion struct {
+	Major     string `json:"major"`
+	Minor     string `json:"minor"`
+	SubMinor  string `json:"sub_minor"`
+	Dev       bool   `json:"dev,omitempty"`
+	GitCommit string `json:"git_commit,omitempty"`
+}
+
+// WriteVersionFile writes the version information to a given file
+// file is the location of the old version file
+// gitCommit is the current git commit version. It will be added to the file
+// to aid in debugging, but will not be used to compare versions
+func WriteVersionFile(file string, gitCommit string) error {
+	currentVersion := parseVersionConstant(gitCommit)
+	jsonVersion, err := json.Marshal(&currentVersion)
+	// This should never happen
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(file, jsonVersion, 0644)
```

Comment:
Let's write this file using https://github.com/google/renameio. It ensures atomic writes as well as fsync.

Reply:
ooh fun new package